### PR TITLE
[no ticket][risk=no]Find out why waitUntilNoSpinner fails in CircleCI jobs

### PR DESCRIPTION
**Pull request by aweng98**
Speculative fix and log to prove works: Increase timeout to 90 seconds and capture screenshot if failed. 
Reason: circleci jobs experiencing random test failures due to spinner timeout. 

```
TimeoutError: waiting for function failed: timeout 60000ms exceeded

      165 |     const selectr2 = 'svg[style*="spin"], .spinner:empty';
      166 |     if (jValue) {
    > 167 |       await this.page.waitFor((selector) => {
          |                       ^
      168 |         return document.querySelectorAll(selector).length === 0
      169 |       }, {timeout: 60000}, selectr2);
      170 |     }

      at new WaitTask (node_modules/puppeteer/lib/DOMWorld.js:549:28)
      at DOMWorld.waitForFunction (node_modules/puppeteer/lib/DOMWorld.js:454:12)
      at Frame.waitForFunction (node_modules/puppeteer/lib/FrameManager.js:657:28)
      at Frame.waitFor (node_modules/puppeteer/lib/FrameManager.js:617:19)
      at Page.waitFor (node_modules/puppeteer/lib/Page.js:1113:29)
      at HomePage.<anonymous> (app/authenticated-page.ts:167:23)
      at fulfilled (node_modules/tslib/tslib.js:110:62)
```

### CODE REVIEW

File: e2e/app/authenticated-page.ts [hunk 1]
```diff
@@ -161,12 +161,28 @@ export default abstract class AuthenticatedPage extends BasePage {
...
     const selectr2 = 'svg[style*="spin"], .spinner:empty';
-    if (jValue) {
-      await this.page.waitFor((selector) => {
-        return document.querySelectorAll(selector).length === 0
-      }, {timeout: 60000}, selectr2);
+    const startTime = performance.now();
+    try {
+      if (jValue) {
+        await this.page.waitFor((selector) => {
+          const len = document.querySelectorAll(selector).length;
```
Comment (NehaBroad) [inline e2e/app/authenticated-page.ts]: nit: change i to selectorLength

File: e2e/app/authenticated-page.ts [hunk 2]
```diff
@@ -161,12 +161,28 @@ export default abstract class AuthenticatedPage extends BasePage {
...
+        }, {timeout: 90000}, selectr2);
+      }
+    } catch (err) {
+      await this.takeScreenshot('TimedOutWaitForSpinnerStop');
+      throw err;
+    } finally {
+      const finishTime = performance.now();
+      const diff = Math.floor(((finishTime - startTime) / 1000) % 60);
+      if (diff > 60) {
+        // if timeout exceeds 60 seconds without error thrown, it shows page loading is slow
```
Comment (NehaBroad) [inline e2e/app/authenticated-page.ts]: should this be 60 or 90?

Comment (aweng98) [reply e2e/app/authenticated-page.ts]: it's correct.

File: e2e/app/authenticated-page.ts (hunk 1 above)
Comment (aweng98) [reply e2e/app/authenticated-page.ts]: 👍

**Pull request merged**